usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -19,89 +19,46 @@
-            except Exception:
-                logger.exception(format_log("", name=task_id, action="Delete"))
-                raise SystemExit(1)
-
-    end_ = client.count(index=index)["count"]
-    record_count = start_ - end_
-    duration = perf_counter() - start
-    msg = f"Delete operation took {duration:.2f}s. Removed {record_count:,} document{'s' if record_count != 1 else ''}"
-    logger.info(format_log(msg, name=task_id, action="Delete"))
-    return
-
-
-def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list, task_id: str, index) -> int:
+    key: str,
+    value_list: list,
+    task_id: str,
+    index,
+    refresh_after: bool = True,
+    delete_chunk_size: int = 10000,
+) -> int:
     """
     Bulk delete a batch of documents whose field identified by ``key`` matches any value provided in the
     ``values_list``.
 
+    NOTE: This delete routine looks at just the index name given. If there are duplicate records across
+    multiple indexes, an alias or wildcard should be provided for ``index`` param that covers multiple
+    indices, or this will need to be run once per index.
+
     Args:
         client (Elasticsearch): elasticsearch-dsl client for making calls to an ES cluster
-        key (str): name of filed in targeted elasticsearch index that should have a unique value for
-            every doc in the index. Ideally the field or sub-field provided is of ``keyword`` type.
+        key (str): name of field in targeted elasticsearch index that should have a unique value for
+            every doc in the index. The field or sub-field provided MUST be of ``keyword`` type (or ``_id`` meta field)
         value_list (list): if key field has these values, the document will be deleted
         task_id (str): name of ES ETL job being run, used in logging
         index (str): name of index (or alias) to target for the ``_delete_by_query`` ES operation.
-
-            NOTE: This delete routine looks at just the index name given. If there are duplicate records across
-            multiple indexes, an alias or wildcard should be provided for ``index`` param that covers multiple
-            indices, or this will need to be run once per index.
+        refresh_after (bool): Whether to call ``_refresh`` on the index when all of the provided values in
+            ``value_list`` have been processed for delete; defaults to ``True``. If many small deletes happen at a
+            rapid rate, it may be best to set this ``False`` and await a deferred refresh afterward in the calling
+            code. However it is important to not retry deletes that have already been deleted until _refresh is
+            called, or a elasticsearch.exceptions.ConflictError will be raised due to an ES doc version_conflict
+            error response. NOTE: This param will be ignored and a refresh will be attempted if this function
```

Comment:
> However it is important to not retry deletes that have already been deleted until _refresh is called, or a elasticsearch.exceptions.ConflictError will be raised due to an ES doc version_conflict error response

Is this still true with `conflicts="proceed"` used below?

Reply:
Nope, removed that part of the comment.